server/server.go

```diff
@@ -455,8 +476,13 @@ func (s *Server) Run(ctx context.Context) error {
 
 	go func() {
 		s.logger.Info(fmt.Sprintf("HTTP server listening on '%s'...", httpServer.Addr))
-		err := httpServer.ListenAndServe()
 
+		var err error
+		if s.config.TLSCertPath != "" && s.config.TLSKeyPath != "" {
```

Comment:
We assume `TLSCertPath` and `TLSKeyPath` are not empty if `TLSEnabled`. If this is not true, then the caller should have validated that beforehand.

So code that uses the TLS configurations can simply change to `if s.config.TLSEnabled {...}`, right?